public/components/datasources/components/manage/data_connection.tsx

```diff
@@ -94,6 +106,23 @@ export const DataConnection = (props: any) => {
   const onclickDiscoverCard = () => {
     application!.navigateToApp(observabilityLogsID);
   };
+  const [isFirstTimeLoading, setIsFirstTimeLoading] = useState<boolean>(true);
+  const {
+    loadStatus: databasesLoadStatus,
+    startLoading: startLoadingDatabases,
+  } = useLoadDatabasesToCache();
+  const { loadStatus: tablesLoadStatus, startLoading: startLoadingTables } = useLoadTablesToCache();
+  const {
+    loadStatus: accelerationsLoadStatus,
+    startLoading: startLoadingAccelerations,
+  } = useLoadAccelerationsToCache();
+  const [tablesIsLoading, setTablesIsLoading] = useState<boolean>(false);
+  const [accelerationsIsLoading, setAccelerationsIsLoading] = useState<boolean>(false);
+  const [cachedDatabases, setCachedDatabases] = useState<CachedDatabase[]>([]);
+  const [cachedTables, setCachedTables] = useState<CachedTable[]>([]);
+  const [cachedAccelerations, setCachedAccelerations] = useState<CachedAcceleration[]>([]);
+  const [selectedDatabase, setSelectedDatabase] = useState<string>('');
+  const [isRefreshing, setIsRefreshing] = useState<boolean>(false);
```

Comment:
we can merge all the loading booleans into a single state containing an object. I feel that is easier to read than this. Something similar to this: https://github.com/opensearch-project/dashboards-observability/blob/41214c22a67157b7a2f1d100162b99d4d6d5d498/public/components/datasources/components/manage/accelerations/selectors/source_selector.tsx#L44-L48